do

```diff
@@ -15,211 +15,450 @@ from epimodel.gleam import Batch, batch
         d.loc[
             ((d.Level == Level.country) | (d.Level == Level.continent))
             & (d.GleamID != "")
         ].Region.values
     )
     # Add all configured regions
-    for rc in args.config["export_regions"]:
-        r = args.rds[rc]
+    for rc in ctx.obj["CONFIG"]["export_regions"]:
+        r = ctx.obj["RDS"][rc]
         if r.GleamID != "":
             regions.add(r)
 
-    log.info(
-        f"Importing results for {len(regions)} from GLEAM into {args.BATCH_FILE} ..."
-    )
+    log.info(f"Importing results for {len(regions)} from GLEAM into {batch_file} ...")
     b.import_results_from_gleam(
-        Path(args.config["gleamviz_sims_dir"]).expanduser(),
+        Path(ctx.obj["CONFIG"]["gleamviz_sims_dir"]).expanduser(),
         regions,
-        resample=args.config["gleam_resample"],
-        allow_unfinished=args.allow_missing,
-        overwrite=args.overwrite,
+        resample=ctx.obj["CONFIG"]["gleam_resample"],
+        allow_unfinished=allow_missing,
+        overwrite=overwrite,
         info_level=logging.INFO,
     )
 
 
-def generate_batch(args):
-    b = Batch.new(dir=args.config["output_dir"], comment=args.comment)
+@action.command()
+@click.argument("base_def", type=click.Path(exists=True))
+@click.argument("country_estimates", type=click.Path(exists=True))
+@click.option(
+    "-t", "--top", default=2000, type=int, help="Upper limit for seed compartments."
+)
+@click.option("-c", "--comment", type=str, help="A short comment (to be part of path).")
+@click.option(
+    "-D",
+    "--start-date",
+    type=click.DateTime(),
+    help="Set a sim start date (default: from the simulation def).",
+)
+@click.pass_context
+def generate_gleam_batch(ctx, base_def, country_estimates, top, comment, start_date):
+    """
+    Create batch of definitions for GLEAM.
+
+    Saved in the directory specified by 'output_dir' in config.yml.
+    
+    BASE_DEF: Basic definition file to use.
+
+    COUNTRY_ESTIMATES: The country-level estimate source CSV file.
+    """
+    b = Batch.new(dir=ctx.obj["CONFIG"]["output_dir"], comment=comment)
     log.info(f"New batch file {b.path}")
-    log.info(f"Reading base GLEAM definition {args.BASE_DEF} ...")
-    d = epimodel.gleam.GleamDefinition(args.BASE_DEF)
-    # TODO: This shuld be somewhat more versatile
-    log.info(f"Reading estimates from CSV {args.COUNTRY_ESTIMATES} ...")
-    est = read_csv_smart(args.COUNTRY_ESTIMATES, args.rds, levels=Level.country)
-    if args.start_date:
-        start_date = utils.utc_date(args.start_date)
+    log.info(f"Reading base GLEAM definition {base_def} ...")
+    d = epimodel.gleam.GleamDefinition(base_def)
+    # TODO: This should be somewhat more versatile
+    log.info(f"Reading estimates from CSV {country_estimates} ...")
+    est = read_csv_smart(country_estimates, ctx.obj["RDS"], levels=Level.country)
+    if start_date:
+        start_date = utils.utc_date(start_date)
     else:
         start_date = d.get_start_date()
```

Comment:
also minor suggestion, feel free to ignore:

```suggestion
    start_date = utils.utc_date(start_date) if start_date else d.get_start_date()
```

Reply:
Made the change :)